Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp

```diff
@@ -56,6 +61,49 @@ const Geometry::BoundingBox &MCInteractionVolume::getBoundingBox() const {
   return m_sample->getBoundingBox();
 }
 
+/**
+ * Generate point randomly across one of the components of the environment
+ * including the sample itself in the selection. The method first selects
+ * a random component and then selects a random point within that component
+ * using Object::generatePointObject
+ * @param rng A reference to a PseudoRandomNumberGenerator where
+ * nextValue should return a flat random number between 0.0 & 1.0
+ * @return The generated point
+ */
+Kernel::V3D MCInteractionVolume::generatePoint(
+    Kernel::PseudoRandomNumberGenerator &rng) const {
+  for (size_t i = 0; i < m_maxScatterAttempts; i++) {
+    Kernel::V3D point;
+    int componentIndex;
+    if (m_env) {
+      componentIndex = rng.nextInt(0, static_cast<int>(m_env->nelements())) - 1;
```

Comment:
Why -1, could we add a comment explaining this magic value?

Reply:
Yes sure